homeassistant/components/miele/binary_sensor.py

```diff
@@ -0,0 +1,283 @@
+"""Binary sensor platform for Miele integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Final, cast
+
+from pymiele import MieleDevice
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import MieleAppliance
+from .coordinator import MieleConfigEntry
+from .entity import MieleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class MieleBinarySensorDescription(BinarySensorEntityDescription):
+    """Class describing Miele binary sensor entities."""
+
+    value_fn: Callable[[MieleDevice], StateType]
```

Comment:
```suggestion
    value_fn: Callable[[MieleDevice], bool]
```

Reply:
See below. The API delivers a StateType today.

Reply:
what type is the StateType?

Reply:
`type StateType = str | int | float | None`. It is defined in homeassistant.helpers.typing and is widely used in the codebase for this purpose.